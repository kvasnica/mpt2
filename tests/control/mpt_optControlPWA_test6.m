function mpt_optControlPWA_test6

% tests that we properly switch to different method when no QP solver is
% available:

% set up CFTOC problem
pwa1d
probStruct.norm = 1;
probStruct.subopt_lev = 0;
probStruct.N = 2;

ctrl = mpt_control(sysStruct, probStruct, struct('qpsolver', -1));
mbg_assertequal(length(ctrl), 6);
