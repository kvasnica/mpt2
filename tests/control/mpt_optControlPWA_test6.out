WARNING: older version of mpLP solver specified, switching to different algorithm...
--- Step 1 (Horizon 2) ---
exploring dynamics 1         
exploring dynamics 2         
4 regions in 2 feasible transitions
Regions: 4
--- Step 2 (Horizon 1) ---
exploring dynamics 1         
exploring dynamics 2         
12 regions in 6 feasible transitions
Removed 1 duplicate regions
Removing overlaps...
hull 1/2	hull 2/2	
Removed 1 duplicate regions
Removing overlaps...
hull 1/2	hull 2/2	
Regions: 6

Solution consists of 6 regions
